app/code/Magento/ReviewGraphQl/Model/Resolver/CreateProductReview.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\ReviewGraphQl\Model\Resolver;
+
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Exception\GraphQlNoSuchEntityException;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+use Magento\Framework\GraphQl\Query\Resolver\Value;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Review\Helper\Data as ReviewHelper;
+use Magento\ReviewGraphQl\Mapper\ReviewDataMapper;
+use Magento\ReviewGraphQl\Model\Review\AddReviewToProduct;
+use Magento\Store\Api\Data\StoreInterface;
+
+/**
+ * Create product review resolver
+ */
+class CreateProductReview implements ResolverInterface
+{
+    /**
+     * @var ReviewHelper
+     */
+    private $reviewHelper;
+
+    /**
+     * @var AddReviewToProduct
+     */
+    private $addReviewToProduct;
+
+    /**
+     * @var ReviewDataMapper
+     */
+    private $reviewDataMapper;
+
+    /**
+     * @param AddReviewToProduct $addReviewToProduct
+     * @param ReviewDataMapper $reviewDataMapper
+     * @param ReviewHelper $reviewHelper
+     */
+    public function __construct(
+        AddReviewToProduct $addReviewToProduct,
+        ReviewDataMapper $reviewDataMapper,
+        ReviewHelper $reviewHelper
+    ) {
+
+        $this->addReviewToProduct = $addReviewToProduct;
+        $this->reviewDataMapper = $reviewDataMapper;
+        $this->reviewHelper = $reviewHelper;
+    }
+
+    /**
+     * Resolve product review ratings
+     *
+     * @param Field $field
+     * @param ContextInterface $context
+     * @param ResolveInfo $info
+     * @param array|null $value
+     * @param array|null $args
+     *
+     * @return array[]|Value|mixed
+     *
+     * @throws GraphQlInputException
+     * @throws GraphQlNoSuchEntityException
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $input = $args['input'];
+        $customerId = null;
+
+        if ($context->getExtensionAttributes()->getIsCustomer()) {
+            $customerId = (int) $context->getUserId();
+        }
+
+        if (!$customerId && !$this->reviewHelper->getIsGuestAllowToWrite()) {
+            throw new GraphQlInputException(__('Guest customers aren\'t allowed to add product reviews.'));
```

Comment:
Looks like should be `GraphQlAuthorizationException`.

Reply:
That makes sense. ✔️
Thank you.